lib/src/http/auto_retry.dart

```diff
@@ -170,18 +171,60 @@ class RetryBackOff {
   /// The back-off method to use. One of none, fixed, or exponential.
   final RetryBackOffMethod method;
 
+  /// Whether to enable jitter or not.
+  final bool enableJitter;
+
+  /// The maximum duration between retries.
+  final maxDurationInMs;
+
+  /// The default maximum duration between retries. (5 minutes)
+  static const defaultMaxDurationInMs = 5 * 60 * 1000;
+
+  int _baseBackOffFunction(int numAttempts) {
+    return this.duration.inMilliseconds * pow(2, numAttempts);
+  }
+
+  Duration calculate(int numAttempts) {
+    Random random = new Random();
+    if (enableJitter != null && enableJitter == true) {
+      if (maxDurationInMs != null) {
+        return new Duration(
+            milliseconds: random.nextInt(
+                min(this.maxDurationInMs, _baseBackOffFunction(numAttempts))));
+      } else {
+        return new Duration(
+            milliseconds: random.nextInt(_baseBackOffFunction(numAttempts)));
+      }
+    } else {
+      if (maxDurationInMs != null) {
+        return new Duration(
+            milliseconds:
+                min(this.maxDurationInMs, _baseBackOffFunction(numAttempts)));
+      } else {
+        return new Duration(milliseconds: _baseBackOffFunction(numAttempts));
+      }
+    }
+  }
+
   /// Construct a new exponential back-off representation where [duration] is
   /// the base duration from which each delay will be calculated.
-  const RetryBackOff.exponential(this.duration)
-      : method = RetryBackOffMethod.exponential;
+  const RetryBackOff.exponential(this.duration,
+      {bool this.enableJitter, maxDurationInMs})
```

Comment:
Should always use `Duration` - it allows the consumer to decide what unit they want to use. Also `maxDuration` to me isn't very descriptive, duration of what? It's a maximum interval between requests, so maybe `maxInterval` or `maxDelay`.

Also I think I might like `withJitter` as opposed to `enableJitter`.